File: iOS7bookExamples/bk2ch06p270modalPresentationContext/ch19p607modalPresentationContext/ExtraViewController.m


#import "ExtraViewController.h"

@interface ExtraViewController ()

@end

@implementation ExtraViewController

- (IBAction)doButton:(id)sender {
    NSLog(@"presented vc's presenting vc: %@", self.presentingViewController);
    [self.presentingViewController dismissViewControllerAnimated:YES completion:nil];
}

@end

File: iOS9bookExamples/bk2ch06p270modalPresentationContext/ch19p607modalPresentationContext/ExtraViewController.swift


import UIKit

class ExtraViewController : UIViewController {
        
    override func prefersStatusBarHidden() -> Bool {
        return true
    }
    
    @IBAction func doButton (sender : AnyObject) {
        print("presented vc's presenting vc: \(self.presentingViewController)")
        self.presentingViewController!.dismissViewControllerAnimated(true, completion: nil)
    }
}
